src/main/java/org/folio/search/service/consortium/ConsortiumSearchHelper.java

```diff
@@ -187,14 +189,35 @@ private Optional<TermQueryBuilder> getBrowseTenantFilter(BrowseContext context)
 
   public static Optional<TermQueryBuilder> getBrowseFilter(BrowseContext context, String filterKey) {
     return context.getFilters().stream()
-      .map(filter ->
-        filter instanceof TermQueryBuilder termFilter && termFilter.fieldName().equals(filterKey)
-        ? termFilter
-        : null)
+      .map(filter -> getTermFilterForKey(filter, filterKey))
       .filter(Objects::nonNull)
       .findFirst();
   }
 
+  public static List<Object> getBrowseFilterValues(BrowseContext context, String filterKey) {
+    return context.getFilters().stream()
+      .flatMap(filter -> getTermFiltersForKey(filter, filterKey))
+      .filter(Objects::nonNull)
+      .map(TermQueryBuilder::value)
+      .toList();
+  }
+
+  private static TermQueryBuilder getTermFilterForKey(QueryBuilder filter, String filterKey) {
+    return filter instanceof TermQueryBuilder termFilter && termFilter.fieldName().equals(filterKey)
+      ? termFilter
+      : null;
+  }
+
+  private static Stream<TermQueryBuilder> getTermFiltersForKey(QueryBuilder filter, String filterKey) {
+    if (filter instanceof TermQueryBuilder termFilter && termFilter.fieldName().equals(filterKey)) {
+      return Stream.of(termFilter);
+    } else if (filter instanceof BoolQueryBuilder boolFilter) {
+      return boolFilter.should().stream().map(shouldFilter -> getTermFilterForKey(shouldFilter, filterKey));
+    }
+    return null;
```

Comment:
```Stream.empty()``` could be returned as I think ```.flatMap()``` in any way would convert to empty stream if mapper returns ```null``` (according to javadoc), or I think there is no need to ```.filter(Objects::nonNull)```

Also, ```getTermFilterForKey(...)``` could also be changed to return Stream